### ISSUE TITLE
Unable to execute transactions on Free Gas Networks

### ISSUE THREAD

**Issue by lukeskywlkr** [MetaMask/metamask-mobile#1890]
**Describe the bug**
The application is unable to execute transactions with gas price set to 0 on free gas networks. Metamask reports 0 as invalid number.

**Screenshots**
![IMG_8030](https://user-images.githubusercontent.com/23049540/95756435-0f8dfc80-0ca6-11eb-8086-630e90546010.jpg)

**To Reproduce**
_Steps to reproduce the behavior_
1. Connect to a Free Gas Network
2. Execute a transaction and try to set the gas price to 0

**Expected behavior**
The app should allow the execution of a transaction on a free gas network and let the user modify accordingly the advanced transaction fee settings.

**Smartphone (please complete the following information):**
 - Device: iPhone6s
 - OS: iOS 14.0.1
 - App Version v1.0.3 (528)

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

**Comment by lukeskywlkr** [comment]
Seems that the error is generated by this check at this line of code: https://github.com/MetaMask/metamask-mobile/blob/19593ceb8a944f7fbc515ee722bfd2f853384067/app/components/Views/SendFlow/CustomGas/index.js#L395

Could it be problematic to allow execution of transactions on such networks?

**Comment by ackhia** [comment]
Seems like this change was made in PR #1750 and has broken compatibility with Quorum and possibly other private network clients.

**Comment by Obafs** [comment]
What advantage was obtained by making this change? In my current use, it prevents mobile users from accessing smart contract methods on private networks, that are available to the desktop version. 

TL;DR: Please allow zero in custom gas price, unless there is a security reason not to.
